src/main/scala/com/advancedtelematic/campaigner/db/ExtractUpdatesFromCampaignsAndInsert.scala

```diff
@@ -0,0 +1,49 @@
+package com.advancedtelematic.campaigner.db
+
+import java.time.Instant
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import com.advancedtelematic.campaigner.data.DataType._
+import com.advancedtelematic.campaigner.db.Schema.{campaigns, updates}
+import com.advancedtelematic.libats.messaging_datatype.DataType.UpdateId
+import com.advancedtelematic.libats.slick.db.SlickExtensions._
+import com.advancedtelematic.libats.slick.db.SlickUUIDKey._
+import org.slf4j.LoggerFactory
+import slick.jdbc.MySQLProfile.api._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class ExtractUpdatesFromCampaignsAndInsert(implicit
+                                           val db: Database,
+                                           val mat: Materializer,
+                                           val system: ActorSystem,
+                                           val ec: ExecutionContext
+                                   ) {
+
+  private val _log = LoggerFactory.getLogger(this.getClass)
+
+  private[db] def updateFromCampaign(c: Campaign): Update = {
+    val u = Update(UpdateId.generate(), UpdateSource(c.updateId.uuid.toString, UpdateType.multi_target), c.namespace, c.name, None, Instant.now, Instant.now)
+    _log.info("Extracted update {} from campaign {}", u, c, "")
+    u
+  }
+
+  private[db] def createOneUpdateRecordForEachCampaign(): Future[Option[Int]] =
```

Comment:
If I am reading this right, this would be the same as:

```
    db.run(campaigns.distinctOn(_.update).result)
      .map { cs => cs.map(updateFromCampaign) }
      .flatMap { us => db.run(updates ++= us) }
```
?

Also, this works because this table is relatively small, but in the future we should write migrations using `db.stream` + async writes to the database so we don't bring the whole table into memory and write it back to mysql. For example see: https://github.com/advancedtelematic/ota-tuf/blob/e54bd1770fa82594887e19ca1222dc0b85d0be0b/libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/db/SignatureMethodMigration.scala#L33